src/ir/manipulation.h

```diff
@@ -33,14 +33,24 @@ inline OutputType* convert(InputType* input) {
   return output;
 }
 
-// Convenience method for nop, which is a common conversion
+// Convenience methods for certain instructions, which are common conversions
 template<typename InputType> inline Nop* nop(InputType* target) {
-  return convert<InputType, Nop>(target);
+  auto* ret = convert<InputType, Nop>(target);
+  ret->finalize();
```

Comment:
Done

Reply:
I can't do that, because unlike the simplest ones here (`nop`, `unreachable`, and `ref.null`,  other instructions have fields that have not been assigned at the end of `convert` yet. So calling `finalize` on them causes a segmentation fault.